File: GMS map editor/ProjectAssets/Rooms/Tile.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace GMSMapEditor.ProjectAssets.Rooms
{
    public class Tile
    {
        public string bgName = "";
        public string name;
        public string colour = "4294967295";
        public double scaleX = 1;
        public double scaleY = 1;
        public bool locked = false;
        public int x = 0;//room
        public int y = 0;//room
        public int w = 0;
        public int h = 0;
        public int xo = 0;//tileset
        public int yo = 0;//tileset
        public int id = 10000000;
        public int depth = 1000000;

        public Tile()
        {
            Random random = new Random();
            this.name = "inst_" + random.Next(0, 999999999);
        }
    }
}

File: Java/JavaMapEditor/src/xml/Tile.java
package xml;

public class Tile {
    public String bgName = "";
    public String name;
    public String colour = "4294967295";
    public double scaleX = 1;
    public double scaleY = 1;
    public boolean locked = false;
    public int x = 0;//room
    public int y = 0;//room
    public int w = 0;
    public int h = 0;
    public int xo = 0;//tileset
    public int yo = 0;//tileset
    public int id = 10000000;
    public int depth = 1000000;
    @Override
    public Tile clone(){
        Tile t = new Tile();
        t.bgName = bgName;
        t.name = name;
        t.colour = colour;
        t.scaleX = scaleX;
        t.scaleY = scaleY;
        t.locked = locked;
        t.x = x;//room
        t.y = y;//room
        t.w = w;
        t.h = h;
        t.xo = xo;//tileset
        t.yo = yo;//tileset
        t.id = id;
        return t;
    }
    
}
